### PR TITLE
Do not start master when one is running

### PR DESCRIPTION
**Pull request by liuhongtong**
Signed-off-by: liuhongtong <hongtongliu@126.com>

### CODE REVIEW

**Comment by LuQQiu** [comment]
@liuhongtong The old behavior of `./bin/alluxio-start.sh master` is 
(1) if there is any existing `AlluxioMaster`, kill the current `AlluxioMaster`.
(2) and then start the new `AlluxioMaster`.

The changes in your PR is when `./bin/alluxio-start.sh -N master`, 
(1) Do not kill existing `AlluxioMaster`
(2) if there is already one existing `AlluxioMaster`, do not start a new one. <-- I think it make sense to maintain the current old `AlluxioMaster`.

What will happen before? We will have two `AlluxioMaster` or the later `AlluxioMaster` cannot be started correctly?

File: bin/alluxio-start.sh [hunk 1]
```diff
@@ -242,6 +242,12 @@ start_master() {
   fi
 
   if [[ ${ALLUXIO_MASTER_SECONDARY} == "true" ]]; then
+    local keyword="alluxio.master.AlluxioSecondaryMaster"
+    pids=($(ps -Aww -o pid,command | grep -i "[j]ava" | grep ${keyword} | awk '{print $1}'))
+    if [ ${#pids[@]} -ge 1 ]; then
+        echo "secondary master is already running @ $(hostname -f)."
```
Comment (LuQQiu) [inline bin/alluxio-start.sh]: User facing messages need to start with capital letter. Changed to something similar to`Secondary master @(hostname -f) is already running`

File: bin/alluxio-start.sh [hunk 2]
```diff
@@ -257,6 +263,12 @@ start_master() {
      ${ALLUXIO_SECONDARY_MASTER_JAVA_OPTS} \
      alluxio.master.AlluxioSecondaryMaster > ${ALLUXIO_LOGS_DIR}/secondary_master.out 2>&1) &
   else
+    local keyword="alluxio.master.AlluxioMaster"
+    pids=($(ps -Aww -o pid,command | grep -i "[j]ava" | grep ${keyword} | awk '{print $1}'))
+    if [ ${#pids[@]} -ge 1 ]; then
+        echo "master is already running @ $(hostname -f)."
```
Comment (LuQQiu) [inline bin/alluxio-start.sh]: Same as previous comment

**Comment by liuhongtong** [comment]
@LuQQiu when ./bin/alluxio-start.sh -N master, a new AlluxioMaster will start, but fail immediately, because the port has been used.

**Comment by LuQQiu** [comment]
@apc999 PTAL, thanks

**Comment by liuhongtong** [comment]
@apc999 PTAL, thanks.
Tests run: 16, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 48.77 sec <<< FAILURE! - in alluxio.server.ft.journal.ufs.UfsJournalIntegrationTest
pin(alluxio.server.ft.journal.ufs.UfsJournalIntegrationTest)  Time elapsed: 1.506 sec  <<< ERROR!
java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: EOF while parsing checkpoint type. Was your checkpoint written by alluxio-1.x? See https://docs.alluxio.io/os/user/edge/en/operation/2.x-Upgrade.html for instructions on how to upgrade from alluxio-1.x to alluxio-2.x
	at alluxio.server.ft.journal.ufs.UfsJournalIntegrationTest.createFsMasterFromJournal(UfsJournalIntegrationTest.java:634)
	at alluxio.server.ft.journal.ufs.UfsJournalIntegrationTest.pinTestUtil(UfsJournalIntegrationTest.java:394)
	at alluxio.server.ft.journal.ufs.UfsJournalIntegrationTest.pin(UfsJournalIntegrationTest.java:390)
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: EOF while parsing checkpoint type. Was your checkpoint written by alluxio-1.x? See https://docs.alluxio.io/os/user/edge/en/operation/2.x-Upgrade.html for instructions on how to upgrade from alluxio-1.x to alluxio-2.x
	at alluxio.server.ft.journal.ufs.UfsJournalIntegrationTest.createFsMasterFromJournal(UfsJournalIntegrationTest.java:634)
	at alluxio.server.ft.journal.ufs.UfsJournalIntegrationTest.pinTestUtil(UfsJournalIntegrationTest.java:394)
	at alluxio.server.ft.journal.ufs.UfsJournalIntegrationTest.pin(UfsJournalIntegrationTest.java:390)
Caused by: java.lang.IllegalStateException: EOF while parsing checkpoint type. Was your checkpoint written by alluxio-1.x? See https://docs.alluxio.io/os/user/edge/en/operation/2.x-Upgrade.html for instructions on how to upgrade from alluxio-1.x to alluxio-2.x
Caused by: java.io.EOFException

**Comment by liuhongtong** [comment]
@apc999 PTAL. Thanks.

**Comment by ZacBlanco** [comment]
Hi @liuhongtong !

We appreciate your contribution, however after some offline discussion with @LuQQiu we would like to see a few things before merging this PR

1. Can we extend this functionality across all alluxio processes, and ensure that the check only runs when the `-N` flag is supplied?
2. Instead of copying and pasting the same functionality, can we abstract out the `s -Aww -o pid,command | grep -i "[j]ava" | grep ${keyword} | awk '{print $1}'` into another function which can be called for each process?

**Comment by liuhongtong** [comment]
@ZacBlanco I accept. But could I commit a new PR to implement your proposal？

**Comment by ZacBlanco** [comment]
Yes, whatever you would like to do. Either open a new PR or continue working on this one.

**Comment by liuhongtong** [comment]
@ZacBlanco Please help to merge this PR. And I will open a new PR.

**Comment by ZacBlanco** [comment]
I don't think that merging this PR first is necessary. You can either open a new PR with this branch as the base or just continue working on top of what you have here. For anything to be merged we would like the previous comments I mentioned addressed.

**Comment by ZacBlanco** [comment]
Since there's been no progress here I'm going to close the PR. 

Please re-open if you would like to continue the work.

**Pull request closed without merging**